app/components/AsciidocBlocks/Document.tsx

```diff
@@ -5,173 +5,23 @@
  *
  * Copyright Oxide Computer Company
  */
+import { useDelegatedReactRouterLinks } from '@oxide/design-system/components/dist'
+import { Content, type DocumentBlock } from '@oxide/react-asciidoc'
+import { useRef } from 'react'
 
-import { Content, type AdocTypes } from '@oxide/react-asciidoc'
-import { Link, useLocation } from '@remix-run/react'
-import { useCallback, useEffect, useMemo, useRef, useState } from 'react'
-import tunnel from 'tunnel-rat'
-
-import { isTruthy } from '~/utils/isTruthy'
-
-import Container from '../Container'
-import { GotoIcon } from '../CustomIcons'
-import {
-  DesktopOutline,
-  generateTableOfContents,
-  SmallScreenOutline,
-  useActiveSectionTracking,
-  useIntersectionObserver,
-} from '../TableOfContents'
-
-export const ui = tunnel()
-
-const CustomDocument = ({ document }: { document: AdocTypes.Document }) => {
```

Comment:
Nice to move most of this out into shared code in `@oxide/design-system` – since we use the ToC on the docs site also.